projects/step-core/src/lib/components/modal-window/modal-window.component.ts

```diff
@@ -1,13 +1,39 @@
-import { Component, Input } from '@angular/core';
+import { AfterViewInit, Component, ElementRef, Input, OnDestroy } from '@angular/core';
 import { MatDialogRef } from '@angular/material/dialog';
 
 @Component({
   selector: 'step-modal-window',
   templateUrl: './modal-window.component.html',
   styleUrls: ['./modal-window.component.scss'],
 })
-export class ModalWindowComponent {
+export class ModalWindowComponent implements AfterViewInit, OnDestroy {
   @Input() title: string = '';
 
-  constructor(public _dialogRef: MatDialogRef<unknown>) {}
+  /**
+   * Sometimes it's require to show the migrated modal above the legacy one
+   * This input can be used to specify the z-index for .cdk-overlay-container when modal appears
+   * As soon as all modals will be migrated, this logic should be removed
+   */
+  @Input() zIndex?: number;
+  private cdkOverlayContainer?: HTMLElement;
+  constructor(private _elRef: ElementRef<HTMLElement>, public _dialogRef: MatDialogRef<unknown>) {}
+
+  ngAfterViewInit(): void {
```

Comment:
this solution looks nice, however it only solves the case with dialogs, and we actually have to solve this problem for all material things that make use of the overlay - an actual example is when we have a mat-select inside an old dialog, the menu with the options gets hidden behind the dialog..

a solution could be to create a directive where this logic is encapsulated and then add that directive to the providers of components that encounter such a case

Reply:
z-index assign logic was moved to a separate directive